File: Algorithm/baekjoon/java/BJ2875.java
package baekjoon.java;

/**
 * Created by changmin on 2018. 1. 10..
 */
import java.util.*;
public class BJ2875 {
    public static int teamMatching(int girls, int boys, int intern){
        int team = 0;
        while(girls >= 2 && boys >= 1 && girls + boys >= intern + 3){
            team += 1;
            girls -= 2;
            boys -= 1;
        }

        return team;
    }
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int girls = sc.nextInt();
        int boys = sc.nextInt();
        int interns = sc.nextInt();
        int output = teamMatching(girls, boys, interns);
        System.out.println(output);
    }
}

File: Algorithm/baekjoon/swift/BJ2875.swift
//2875
func teamMatching(n: Int, m: Int, interns: Int) -> Int{
    var team = 0
    var girl = n
    var boy = m
    while girl >= 2 && boy >= 1 && girl + boy >= interns + 3{
        team += 1;
        girl -= 2;
        boy -= 1;
    }
    return team
}

var input = readLine()
if let input = input{
    var inputs = input.split(separator: " ")
    let girls = Int(inputs[0])!
    let boys = Int(inputs[1])!
    let interns = Int(inputs[2])!
    let output = teamMatching(n: girls, m: boys, interns: interns)
    print(output)
}
